src/flamegpu/io/Telemetry.cpp

```diff
@@ -0,0 +1,167 @@
+
+    // OS
+#ifdef _WIN32
+    payload_items["operatingSystem"] = "Windows";
+#elif __GNUC__ >= 4
+    payload_items["operatingSystem"] = "Unix";
+#else
+    payload_items["operatingSystem"] = "Other";
+#endif
+    // VIS
+#ifdef VISUALISATION
+    payload_items["Visualisation"] = "true";
+#else
+    payload_items["Visualisation"] = "false";
+#endif
+
+    // create the payload
+    std::string payload = "";
+    bool first = true;
+    // iterate payload to generate the payload array
+    for (const auto& [key, value] : payload_items) {
+        std::string item_str;
+        if (!first)
+            item_str = ",\"" + key + ":" + value + "\"";
+        else
+            item_str = "\"" + key + ":" + value + "\"";
+        payload.append(item_str);
+        first = false;
+    }
+
+    // create the telkemtry json package
+    std::string telemetry_data = R"json(
+    [{
+        "isTestMode": "$TEST_MODE",
+        "appID": "$APP_ID",
+        "clientUser": "$BUILD_HASH",
+        "sessionID": "",
+        "type" : "$EVENT_TYPE",
+        "payload" : [$PAYLOAD]
+    }])json";
+    // update the placeholders
+    telemetry_data.replace(telemetry_data.find(var_testmode), var_testmode.length(), testmode);
+    telemetry_data.replace(telemetry_data.find(var_appID), var_appID.length(), appID);
+    telemetry_data.replace(telemetry_data.find(var_buildHash), var_buildHash.length(), buildHash);
+    telemetry_data.replace(telemetry_data.find(var_eventName), var_eventName.length(), event_name);
+    telemetry_data.replace(telemetry_data.find(var_payload), var_payload.length(), payload);
+    // murder the formatting into a single compact line
+    telemetry_data.erase(std::remove(telemetry_data.begin(), telemetry_data.end(), '\n'), telemetry_data.end());  // Remove newlines and replace with space
+    telemetry_data.erase(std::remove(telemetry_data.begin(), telemetry_data.end(), '\t'), telemetry_data.end());  // Remove tabs and replace with space
+    telemetry_data.erase(std::remove_if(telemetry_data.begin(), telemetry_data.end(), [](char c) {
+            return std::isspace(static_cast<unsigned char>(c));
+        }), telemetry_data.end());  // Remove spaces
+    size_t pos = 0;
+    while ((pos = telemetry_data.find("\"", pos)) != std::string::npos) {   // Use escape characters
+        telemetry_data.replace(pos, 1, "\\\"");
+        pos += 2;
+    }
+
+    return telemetry_data;
+}
+
+bool sendTelemetryData(std::string telemetry_data) {
+    // Silent curl command (-s) and redirect response output to null
+    std::string null;
+#if _WIN32
+    null = "nul";
+#else
+    null = "/dev/null";
+#endif
+    std::string curl_command = "curl -s -o " + null + " -X POST \"" + std::string(TELEMETRY_ENDPOINT) + "\" -H \"Content-Type: application/json; charset=utf-8\" --data-raw \"" + telemetry_data + "\"";
+
+    // capture the return value
+    if (std::system(curl_command.c_str()) != EXIT_SUCCESS)
```

Comment:
There is no distinguishing if curl is available or not, or if some other error occured (i.e. a timeout when you have set one)

Reply:
No and there doesn't need to be as both will fail silently.